### DISCUSSION TITLE
PrimitiveDateTime strips leading 0s in hour

### DISCUSSION THREAD

**Discussion by saltedpotatos** [time-rs/time#639] [Question]
When I deserialize a datetime, e.g. `2023-11-30 02:06:15` as a PrimitiveDateTime, it strips the leading 0 and returns 
`2023-11-30 2:06:15.0`. Unfortunately, this is no longer parseable as a PrimitiveDateTime, with a malformed hour field. 

Is there a way to have it not trim that 0?

**Comment by jhpratt** [comment]
Can you provide code to explain what you're doing? It sounds like two unrelated things that have no guarantee of compatibility.

**Reply by saltedpotatos** [reply]
Sure! Here's a playground [link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=dfb1027ed3f5cf860650f5e989215465).

**Reply by jhpratt** [reply]
Ah, you had me thinking it was something entirely different because you said "deserialize" (which implies `serde`).

In any situation, there are two things at work here. You are assuming the output of `.to_string()` is a format that is not, and the current format returned does not contain leading zeroes. The change is simple: you need to alter `[hour]` to `[hour padding:none]`. Full information on the format description can be found [on this page](https://time-rs.github.io/book/api/format-description.html). For clock hour, it says

> The padded value has a width of 2. You can choose between padding with zeroes, spacing, or having no padding at all. The default is to pad the value with zeroes.
